tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,464 @@
+
+    return myEnv
+
+def build_perfharness(coreclrRepo, sandboxDir, extension, dotnetEnv):
+    # Confirm dotnet works
+    dotnet = os.path.join(coreclrRepo, 'Tools', 'dotnetcli', 'dotnet' + extension)
+    runArgs = [dotnet,
+            '--info'
+            ]
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=dotnetEnv)
+    (out, err) = proc.communicate()
+
+    if not proc.returncode == 0:
+        print_error(out, err, 'Failed to get information about the CLI tool.')
+
+    # Restore PerfHarness
+    perfHarnessPath = os.path.join(coreclrRepo, 'tests', 'src', 'Common', 'PerfHarness', 'PerfHarness.csproj')
+    runArgs = [dotnet,
+            'restore',
+            perfHarnessPath]
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=dotnetEnv)
+    (out, err) = proc.communicate()
+
+    if not proc.returncode == 0:
+        print_error(out, err, 'Failed to restore PerfHarness.csproj')
+
+    # Publish PerfHarness
+    runArgs = [dotnet,
+            'publish',
+            perfHarnessPath,
+            '-c',
+            'Release',
+            '-o',
+            sandboxDir]
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=dotnetEnv)
+    (out, err) = proc.communicate()
+
+    if not proc.returncode == 0:
+        print_error(out, err, 'Failed to publish PerfHarness.csproj')
+
+def copytree(src, dst):
+    for item in os.listdir(src):
+        s = os.path.join(src, item)
+        d = os.path.join(dst, item)
+        if os.path.isdir(s):
+            shutil.copytree(s, d)
+        else:
+            shutil.copy2(s, d)
+
+def main(args):
+    if not is_supported_version():
+        log("Python 3.5 or newer is required")
+        return 1
+
+    coreclrPerf, arch, operatingSystem, configuration, jitName, optLevel, runType, outputDir, stabilityPrefix, isScenarioTest, benchviewPath, isPgoOptimized, benchviewGroup, hasWarmupRun, collectionFlags, isLibrary, uploadToBenchview, better, sliceNumber, sliceConfigFile = validate_args(args)
+
+    platform = sys.platform
+    python = 'py'
+    if platform == 'linux' or platform == 'linux2':
+        platform = 'Linux'
+        python = 'python3'
+    elif platform == 'darwin':
+        platform = 'OSX'
+        python = 'python3'
+    elif platform == 'win32':
+        platform = "Windows_NT"
+    else:
+        raise ValueError("Platform %s is not supported" % platform)
+
+    executable = ['cmd.exe', '/c'] if platform == 'Windows_NT' else []
+
+    coreclrRepo = os.getcwd()
+    etwCollection = 'Off' if collectionFlags == 'stopwatch' else 'On'
+    sandboxDir = os.path.join(coreclrRepo, 'bin', 'sandbox')
+    sandboxOutputDir = outputDir if not outputDir is None else os.path.join(sandboxDir, 'Logs')
+
+    extension = '.exe' if platform == 'Windows_NT' else ''
+
+    myEnv = os.environ
```

Comment:
This is reference assignment, and does not copy. Is that the intention? This will result in editing the actual environment variables for the process. If not intended, use `myEnv = dict(os.environ)` to copy